src/main/java/roomescape/service/ReservationService.java

```diff
@@ -1,39 +1,69 @@
 package roomescape.service;
 
 import java.time.LocalDate;
+import java.time.LocalDateTime;
 import java.util.List;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 import roomescape.model.Reservation;
+import roomescape.model.Theme;
 import roomescape.model.TimeSlot;
 import roomescape.repository.ReservationRepository;
+import roomescape.repository.ThemeRepository;
 import roomescape.repository.TimeSlotRepository;
 
 @Service
 public class ReservationService {
 
     private final ReservationRepository reservationRepository;
     private final TimeSlotRepository timeSlotRepository;
+    private final ThemeRepository themeRepository;
 
     @Autowired
     public ReservationService(
         final ReservationRepository reservationRepository,
-        final TimeSlotRepository timeSlotRepository
+        final TimeSlotRepository timeSlotRepository,
+        final ThemeRepository themeRepository
     ) {
         this.reservationRepository = reservationRepository;
         this.timeSlotRepository = timeSlotRepository;
+        this.themeRepository = themeRepository;
     }
 
-    public Reservation reserve(String name, LocalDate date, Long timeId) {
+    public Reservation reserve(final String name, final LocalDate date, final Long timeId, final Long themeId) {
         var timeSlot = findTimeSlot(timeId);
-        var reservation = new Reservation(null, name, date, timeSlot);
+        var theme = findTheme(themeId);
+        var reservation = new Reservation(name, date, timeSlot, theme);
+        validatePastDateTime(reservation);
+        validateDuplicateReservation(reservation);
         var id = reservationRepository.save(reservation);
-        return new Reservation(id, name, date, timeSlot);
+        return new Reservation(id, name, date, timeSlot, theme);
+    }
+
+    private void validatePastDateTime(final Reservation reservation) {
+        if (reservation.isBefore(LocalDateTime.now())) {
+            throw new IllegalArgumentException("이전 날짜로 예약할 수 없습니다.");
+        }
+    }
+
+    private void validateDuplicateReservation(final Reservation reservation) {
+        var reservations = allReservations();
```

Comment:
```suggestion
        var reservations = findAllReservations();
```
자바의 메서드명 컨벤션은 명사가 아닌 동사입니다.

> Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html

Reply:
수정하겠습니다!